File: Carly_Michael/jectacular-ql/src/app/domain/ast/qls/stylesheet.ts
import {Page} from './page';
import {Location} from '../location';
import {QlsNode} from './qls-node';
import {QlsVisitor} from './visitors/qls-visitor';

export class Stylesheet extends QlsNode {
  constructor(readonly name: string, readonly pages: Page[], readonly location: Location) {
    super();
  }

  accept<T>(visitor: QlsVisitor<T>): T {
    return visitor.visitStylesheet(this);
  }
}

File: DogeQL/src/main/kotlin/qls/ast/model/StyleSheet.kt
package qls.ast.model

import qls.ast.node.QlsNode
import qls.visitor.QlsVisitor

data class StyleSheet(val pages: List<Page>, val name: String) : QlsNode {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}